integrations/integrations.go

```diff
@@ -0,0 +1,231 @@
+package integrations
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"syscall"
+
+	cometdb "github.com/cometbft/cometbft-db"
+	cometconfig "github.com/cometbft/cometbft/config"
+	dbm "github.com/cosmos/cosmos-db"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/server"
+	serverconfig "github.com/cosmos/cosmos-sdk/server/config"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+	"github.com/polymerdao/monomer"
+	"github.com/polymerdao/monomer/environment"
+	"github.com/polymerdao/monomer/genesis"
+	"github.com/polymerdao/monomer/node"
+	"github.com/spf13/viper"
+)
+
+const (
+	monomerGenesisPathFlag = "monomer-genesis-path"
+	monomerEngineWSFlag    = "monomer-engine-ws"
+)
+
+// StartCommandHandler is a custom callback that overrides the default `start` function in the Cosmos
+// SDK. It starts a Monomer node in-process instead of a CometBFT node.
+func StartCommandHandler(
+	svrCtx *server.Context,
+	clientCtx client.Context, //nolint:gocritic // hugeParam
+	appCreator servertypes.AppCreator,
+	inProcessConsensus bool,
+	opts server.StartCmdOptions,
+) error {
+	// We assume `inProcessConsensus` is true for now, so let's return an error if it's not.
+	if !inProcessConsensus {
+		return fmt.Errorf("in-process consensus must be enabled")
+	}
+
+	env := environment.New()
+	defer func() {
+		if err := env.Close(); err != nil {
+			svrCtx.Logger.Error("Failed to close environment", "err", err)
+		}
+	}()
+
+	svrCfg, err := serverconfig.GetConfig(svrCtx.Viper)
+	if err != nil {
+		return fmt.Errorf("get server config: %v", err)
+	}
+	if err := svrCfg.ValidateBasic(); err != nil {
+		return fmt.Errorf("validate server config: %v", err)
+	}
+
+	app, err := startApp(env, svrCtx, appCreator, opts)
+	if err != nil {
+		return fmt.Errorf("start application: %v", err)
+	}
+
+	// Would usually start a Comet node in-process here, but we replace the
+	// Comet node with a Monomer node.
+	if err := startInProcess(env, svrCtx, &clientCtx, app, opts); err != nil {
+		return fmt.Errorf("start Monomer node in-process: %v", err)
+	}
+
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
+	<-sigCh
+
+	return nil
+}
+
+var _ io.WriteCloser = (*fakeTraceWriter)(nil)
+
+type fakeTraceWriter struct{}
+
+func (f *fakeTraceWriter) Write([]byte) (int, error) {
+	return 0, nil
+}
+
+func (f *fakeTraceWriter) Close() error {
+	return nil
+}
+
+// See https://github.com/cosmos/cosmos-sdk/blob/7fb26685cd68a6c1d199dc270c80f49f2bfe7ace/server/start.go#L624
+func startApp(
+	env *environment.Env,
+	svrCtx *server.Context,
+	appCreator servertypes.AppCreator,
+	opts server.StartCmdOptions,
+) (app servertypes.Application, err error) {
```

Comment:
Let's avoid named return values unless we need them (I think this is left over the from the Cosmos SDK).
```suggestion
) (servertypes.Application, error) {
```